cmd/sops/main.go

```diff
@@ -966,8 +970,9 @@ func inputStore(context *cli.Context, path string) common.Store {
 	return common.DefaultStoreForPathOrFormat(path, context.String("input-type"))
 }
 
+// SCARY! SHOULD THIS BE USING INPUT-TYPE?
```

Comment:
No, using output-type is correct. Please revert me, or show why you think I'm wrong :D

Reply:
@autrilla I freely-admit that I'm probably poking the wrong part of this code. Here was my (simplistic) test.

With the file test containing only: "`superSecret: superSecretValue`"

and a sops/v3/cmd/sops/main.go containing:

```
func outputStore(context *cli.Context, path string) common.Store {
        //fmt.Println("Doing the right thing")
        //return common.DefaultStoreForPathOrFormat(path, context.String("output-type"))

        fmt.Println("Doing the wrong thing")
        return common.DefaultStoreForPathOrFormat(path, context.String("input-type"))
}
```

I get the following results:

```
# sops exec-file --input-type yaml /tmp/test 'cat {}'
Doing the right thing
Error dumping file: No binary data found in tree
```

```
# sops exec-file --input-type yaml /tmp/test 'cat {}'
Doing the wrong thing
superSecret: superSecretValue
```

Thank you for your time.

Reply:
I see! I think the right thing to do here would be adding support for --output-type too. The issue is that SOPS tries to decide on the output store to use based on the file name, and assumes it should be binary (since `/tmp/test` does not have an extension that indicates a known file type). You'll need to make your calls to sops be of the form `sops exec-file --input-type yaml --output-type yaml /tmp/test 'cat {}'` afterwards.

Reply:
> I think the right thing to do here would be adding support for --output-type too. 

Well that's fair. PR is updated for same.